lib/daru/vector.rb

```diff
@@ -238,9 +238,15 @@ def [](*indexes)
           indexes = indexes.map { |e| named_index_for(e) }
         end
 
-        Daru::Vector.new(
-          indexes.map { |loc| @data[@index[loc]] }, 
-          name: @name, index: indexes, dtype: @dtype)
+        begin
+          Daru::Vector.new(
+            indexes.map { |loc| @data[@index[loc]] },
+            name: @name, index: indexes, dtype: @dtype)
+        rescue NoMethodError
+          raise IndexError, "Specified index #{pos.inspect} does not exist."
+        rescue TypeError
+          raise IndexError, "One or more of specified indices #{pos.inspect} do not exist."
```

Comment:
@v0dro The documentation says:

```ruby
    # == Usage
    #   # For vectors employing single layer Index
    # 
    #   v[:one, :two] # => Daru::Vector with indexes :one and :two
    #   v[:one]       # => Single element
    #   v[:one..:three] # => Daru::Vector with indexes :one, :two and :three
```
It would be counter intuitive if we let `[:x, :y]` fail and let `1..3` as right input as given in the documentation because both `[:x, :y]` and `1..3` are Enumerable. What do you say?